### ISSUE TITLE
coherent single-phonon kernel for powder: probability factor for picking Q vector

### ISSUE THREAD

**Issue by yxqd** [mcvine/mcvine#252]
After [this change](https://github.com/mcvine/mcvine/commit/cc6475cae67dd00d60984c763a7acbfb3e528ba8#diff-0246019f7705a90a10be5294f346e410) the probability factor for picking Q vector seems working for variable Ei. But the [formula used](https://github.com/mcvine/mcvine/blob/cc6475cae67dd00d60984c763a7acbfb3e528ba8/packages/mccomponents/libmccomponents/kernels/sample/phonon/CoherentInelastic_PolyXtal.cc#L248) is a bit obscure. 

Is there a more natural way to calculate this factor?

**Comment by yxqd** [comment]
Using graphite as an example, before the last commit, we have 

![mcvine-sim-graphite-ei140](https://user-images.githubusercontent.com/1796155/33287184-e27277fa-d385-11e7-946c-624a648d6aeb.png)

After the last commit, we got

![mcvine-sim-graphite-ei140-2](https://user-images.githubusercontent.com/1796155/33287233-0bddebe2-d386-11e7-9f12-205c5e6bb9e3.png)

Very similar but the intensities around E=108meV are different.

**Comment by yxqd** [comment]
![image](https://user-images.githubusercontent.com/1796155/33299841-74ff8dda-d3bb-11e7-93b9-07605f0ec762.png)

**Comment by yxqd** [comment]
This probably is not be the right way. The correct way is running another MC simulation beforehand to compute the "accessible dynamical range" like what we did before [the change to use the formula](https://github.com/mcvine/mcvine/commit/cc6475cae67dd00d60984c763a7acbfb3e528ba8#diff-0246019f7705a90a10be5294f346e410), but that method only worked for one Ei. The problem was to compute it for several different Ei is time-consuming, and that was why we switched to the formula.
